util/fetch/src/client.rs

```diff
@@ -291,7 +289,15 @@ impl Fetch for Client {
 			.and_then(|_| rx_res.map_err(|oneshot::Canceled| Error::BackgroundThreadDead))
 			.and_then(future::result);
 
-		Box::new(self.timer.timeout(future, maxdur))
+		Box::new(future.timeout(maxdur)
+			.map_err(|err| {
+				if err.is_inner() {
+					Error::from(err.into_inner().unwrap())
```

Comment:
Handle the error instead of `unwrapping here`

Something like should work:

```rustc
if let Some(inner) = err.into_inner() {
   Error::from(inner)
} else {
   Error::TokioTimeoutInnerVal("".into())
}

```

Reply:
The problem with doing it like that is that [this particular error](https://docs.rs/tokio/0.1.11/tokio/timer/timeout/struct.Error.html) basically has three variants, inner, timer, and elapsed. By calling `into_inner` first, you can't rely on `Error`'s `From` impl to determine which of the other two variants it is.

Basically this one's a bit tricky because in this case I want to use the `From` impl for the inner error type (to get the right variant), not the `timeout::Error` itself.